kubeshop/kusk-gateway
# `.github/workflows/pull-request.yaml`: Rework `e2e` job to use `kind`

Summary
=======

E2E tests / smoketests run using running using `kind`.

Several fixes to make `make check-all` pass locally.

Re-enable `make check-cache` test.

`skaffold.yaml`
--------------

Run `skaffold fix`.

`smoketests/**/*`
-----------------

Several fixes in e2e tests files.

Re-enable `make check-cache` test.

`smoketests/helpers/helpers.go`
-------------------------------

Add helper functions and constants.

Add `GetEnvoyFleetServiceLoadBalancerIP` to get `LoadBalancer`'s IP of `kusk-gateway-envoy-fleet`.

Use config below when installing `kubeshop/kusk-gateway-envoyfleet` and when running e2e tests:

```yaml
  fleet:
    name: kusk-gateway-envoy-fleet
    namespace: kusk-system
```

`.github/workflows/pull-request.yaml`
-------------------------------------

1. Build CRDs.
2. Build `kubeshop/kusk-gateway` (control-plane) image.
3. Install `kubeshop/kusk-gateway` (control-plane) image into cluster. After building `kubeshop/kusk-gateway`, load the
 image into `kind` using `kind load docker-image kubeshop/kusk-gateway...`.
4. Deploy `kusk-gateway-envoy-fleet` (default Envoy Fleet) into cluster.

Inspiration taken from `development/cluster/create-env.sh`.

Replace `e2e` job with a job that runs on a `kind` cluster.

Remove `deploy` step as it is no longer needed, i.e., we are building the images locally in the CI.

Remove `cleanup` step as it is no longer needed, i.e., we are not deploying on the GKE cluster anymore.

Remove `build-manager-docker` as it is no longer needed, i.e., we are not deploying on the GKE cluster anymore.

`smoketests/cluster-config/kind/cluster-configure.sh`
-----------------------------------------------------

Configure `MetalLB` `LoadBalancer` in `kind`.

---

Signed-off-by: Mohamed Bana <mohamed@bana.io>
